File: practice_algos/javascript/algo11.js
// 11. Max/Min/Avg - Given an array with multiple values, write a function
// that returns a new array that only contains the maximum, minimum, and
// average values of the original array. (e.g. [1,5,10,-2] will return
// [10,-2,3.5])

function maxMinAvg(arr) {
    var max = arr[0];
    var min = arr[0];
    var sum = 0;
    for (var i = 0; i < arr.length; i++) {
        if (arr[i] > max) {
            max = arr[i];
        }
        if (arr[i] < min) {
            min = arr[i]
        }
        sum += arr[i];
    }
    var avg =  sum / arr.length;
    return [max, min, avg]
}

console.log(maxMinAvg([1,5,10,-2]))

File: practice_algos/python/algo11.py
# 11. Max/Min/Avg - Given an array with multiple values, write a function
# that returns a new array that only contains the maximum, minimum, and
# average values of the original array. (e.g. [1,5,10,-2] will return
# [10,-2,3.5])

def max_min_avg(arr):
    max = arr[0]
    min = arr[0]
    sum = 0
    for num in arr:
        if num >= max:
            max = num
        if num <= min:
            min = num
        sum += num
    avg = sum / len(arr)
    return [max, min, avg]


print(max_min_avg([1, 5, 10, -2]))
